e2e-samples/GitIntegration-UpdateFromGit.ps1

```diff
@@ -0,0 +1,109 @@
+# 3. Run PowerShell as an administrator
+# 4. Fill in the parameters below
+# 5. Change PowerShell directory to where this script is saved
+# 5. > ./GitIntegration-UpdateFromGit.ps1
+# 6. [Optional] Wait for long running operation to be completed - see LongRunningOperation-Polling.ps1
+
+# Parameters - fill these in before running the script!
+# =====================================================
+
+$workspaceName = "FILL ME"      # The name of the workspace
+
+# End Parameters =======================================
+
+function GetFabricHeaders($resourceUrl) {
+
+    #Login to Azure
+    Connect-AzAccount | Out-Null
+
+    # Get authentication
+    $fabricToken = (Get-AzAccessToken -ResourceUrl $resourceUrl).Token
+
+    $fabricHeaders = @{
+        'Content-Type' = "application/json"
+        'Authorization' = "Bearer {0}" -f $fabricToken
+    }
+
+    return $fabricHeaders
+}
+
+function GetWorkspaceByName($baseUrl, $fabricHeaders, $workspaceName) {
+    # Get workspaces    
+    $getWorkspacesUrl = "{0}/workspaces" -f $baseUrl
+    $workspaces = (Invoke-RestMethod -Headers $fabricHeaders -Uri $getWorkspacesUrl -Method GET).value
+
+    # Try to find the workspace by display name
+    $workspace = $workspaces | Where-Object {$_.DisplayName -eq $workspaceName}
+
+    return $workspace
+}
+
+function GetErrorResponse($exception) {
+    # Relevant only for PowerShell Core
+    $errorResponse = $_.ErrorDetails.Message
+
+    if(!$errorResponse) {
+        # This is needed to support Windows PowerShell
+        $result = $exception.Response.GetResponseStream()
+        $reader = New-Object System.IO.StreamReader($result)
+        $reader.BaseStream.Position = 0
+        $reader.DiscardBufferedData()
+        $errorResponse = $reader.ReadToEnd();
+    }
+
+    return $errorResponse
+}
+
+try {
+    # Set up API endpoints
+    $resourceUrl = "https://api.fabric.microsoft.com"
+    $baseUrl = "$resourceUrl/v1"
+
+    $fabricHeaders = GetFabricHeaders $resourceUrl
+
+    $workspace = GetWorkspaceByName $baseUrl $fabricHeaders $workspaceName 
+    
+    # Verify the existence of the requested workspace
+	if(!$workspace) {
+	  Write-Host "A workspace with the requested name was not found." -ForegroundColor Red
+	  return
+	}
+	
+    # Get Status
+    Write-Host "Calling GET Status REST API to construct the request body for UpdateFromGit REST API."
+
+    $gitStatusUrl = "{0}/workspaces/{1}/git/status" -f $baseUrl, $workspace.Id
+    $gitStatusResponse = Invoke-RestMethod -Headers $fabricHeaders -Uri $gitStatusUrl -Method GET
+
+    # Update from Git
+    Write-Host "Updating the workspace '$workspaceName' from Git has been started."
+
+    $updateFromGitUrl = "{0}/workspaces/{1}/git/updateFromGit" -f $baseUrl, $workspace.Id
+
+    $updateFromGitBody = @{ 
+        remoteCommitHash = $gitStatusResponse.RemoteCommitHash
+		workspaceHead = $gitStatusResponse.WorkspaceHead
```

Comment:
Do we want a lighter version of this sample - getting the HEAD from user (without using GetStatus)?